### DISCUSSION TITLE
Use of cylindrical coordinates simulations with dipole sources on random orientation

### DISCUSSION THREAD

**Discussion by kalosu** [NanoComp/meep#2447] [Q&A]
Hello again meep community,

I have been using meep for doing some simulations in cylindrical coordinates for point-like dipole emitters embedded in some semiconductor layers.

In general I understand that what is needed is the material composition to have cylindrical symmetry for which any illumination configuration can be used as a source field for the simulation.

Is it then possible to simulate for dipole emitters that have a polarization different than er or ep? If I wanted to model the case of having a dipole polarized along z or in any orientation within the rz plane, could I also model this using the cylindrical coordinates implementation?

If you could share with me some literature on the general idea behind the Fourier angular expansion for cylindrical coordinates I think I could better understand how to model such a system. I cannot get to fully understand the concept behind the cylindrical symmetries for the different polarizations.

For instance, why a ez source at r=0 cannot be expanded for m=1 and -1?

Any comments and feedback will be appreciated.

**Comment by smartalecH** [comment]
Have you seen, for example, [how to construct a circularly-polarized source](https://meep.readthedocs.io/en/latest/FAQ/#how-do-i-create-a-circularly-polarized-planewave-source-in-cylindrical-coordinates) in cylindrical coordinates? Or how to solve for the fields as a result of a _linear_ polarization by [summing a superposition of two cylindrical simulations](https://meep.readthedocs.io/en/latest/Scheme_Tutorials/Cylindrical_Coordinates/#scattering-cross-section-of-a-finite-dielectric-cylinder) with different m numbers?

Both of theses links (in particular, the latter) describe the nuances of the m-number you are asking about. 

Note that, if you are trying to find the ensemble average of a series of _random_ sources (eg characterizing the LEE of an LED) there are a number of more efficient and _deterministic_ ways to do so [in meep](https://meep.readthedocs.io/en/latest/Python_Tutorials/Custom_Source/#stochastic-dipole-emission-in-light-emitting-diodes). 

In fact, there's an elegant way to do all this [in cylindrical coordinates too](https://github.com/NanoComp/meep/issues/2108), although there are still some kinks to be worked out...

**Comment by kalosu** [comment]
HI @smartalecH , thanks for the nice feedback.

Yes I have seen those examples in where a linearly polarized plane wave is expressed in cylindrical coordinates. I have also obtained the total field for a dipole polarized along r(sitting at r=0) from the superposition of the m=1 and m=-1 modes. 

However, I would like to test for a dipole source with a polarization that is the linear superposition of a dipole polarized along r and a dipole polarized along z, basically I want to define the polarization vector to be within the rz plane along any angle with respect to the r axis.

If I try using a ez source for my dipole located at r=0 and compute for the m=1,0 and -1 modes I get a message indicating me that for a ez source at r=0 it is not possible to compute for the m=1 or -1 modes. How is this so?

How can I know which azimuthal modes are supported and which ones no? And also if I want to model a dipole with a random orientation, how can I know which modes I should compute for?

Maybe you have some theory reference to this expansion method in cylindrical coordinates that could help me understand more about the symmetry rules for the dipole sources in cylindrical coordinates.
